precompiles/xcm/src/lib.rs

```diff
@@ -181,19 +182,27 @@ where
         input.expect_arguments(6)?;
 
         // Read arguments and check it
-        let assets: Vec<MultiLocation> = input
-            .read::<Vec<Address>>()?
+        let assets_raw: Vec<_> = input
+            .read::<BoundedVec<Address, GetMaxAssets<Runtime>>>()?
+            .into();
+
+        let assets = assets_raw
             .iter()
             .cloned()
             .filter_map(|address| {
                 Runtime::address_to_asset_id(address.into()).and_then(|x| C::reverse_ref(x).ok())
             })
-            .collect();
-        let amounts_raw = input.read::<Vec<U256>>()?;
-        if amounts_raw.iter().any(|x| *x > u128::MAX.into()) {
-            return Err(revert("Asset amount is too big"));
-        }
-        let amounts: Vec<u128> = amounts_raw.iter().map(|x| x.low_u128()).collect();
+            .collect::<Vec<MultiLocation>>();
+
+        let amounts_raw: Vec<U256> = input
+            .read::<BoundedVec<U256, GetMaxAssets<Runtime>>>()?
+            .into();
+
+        let amounts = amounts_raw
+            .iter()
+            .map(|x| (*x).try_into())
```

Comment:
Why does `x` need to be dereferenced here?

Reply:
because of the map() which gives us &U256 value,
the trait `From<&sp_core::U256>` is not implemented for `u128`,  but `From<sp_core::U256>` is  implemented for `u128`.

Reply:
`into_iter` is for such cases.